src/hotspot/share/runtime/os.cpp

```diff
@@ -1190,18 +1189,18 @@ bool os::is_first_C_frame(frame* fr) {
   uintptr_t sp_align_mask = (uintptr_t)(sizeof(int)-1);
 
   uintptr_t usp    = (uintptr_t)fr->sp();
-  if ((usp & sp_align_mask) != 0) return true;
+  if ((usp & sp_align_mask) != 0 || SafeFetchN(fr->sp(), 0) == 0) return true;
```

Comment:
This doesn't quite make sense to me. If the SafeFetchN were to fail then the load in the previous line would already have crashed wouldn't it?

Reply:
Sorry ignore that. The SafeFetch loads `*fr->sp()`.